sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs

```diff
@@ -291,6 +291,70 @@ public virtual Response<DataFeed> CreateDataFeed(string dataFeedName, DataFeedSo
             }
         }
 
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="dataFeedId"></param>
+        /// <param name="dataFeed"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public virtual async Task<Response> UpdateDataFeedAsync(string dataFeedId, DataFeed dataFeed, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNullOrEmpty(dataFeedId, nameof(dataFeedId));
+            Argument.AssertNotNull(dataFeed, nameof(dataFeed));
+            if (!dataFeedId.Equals(dataFeed.Id, StringComparison.OrdinalIgnoreCase))
+            {
+                throw new ArgumentException($"{nameof(dataFeedId)} does not match {nameof(dataFeed.Id)}");
+            }
```

Comment:
There is a good reason for this, but it raises the fact that we need to modify this check to accommodate for these two scenarios. 

The reason is that there are two ways a user may generate an update payload - 
- first get the data feed from the service and then modify it.
- generate a data feed model from scratch, knowing the Id of an existing data feed. 

In the later example, the Id won't be populated because it cannot be set by the user.